bellman/src/groth16/generator/mod.rs

```diff
@@ -0,0 +1,82 @@
+use rand_core::RngCore;
+
+use std::sync::Arc;
+
+use ff::Field;
+use group::CurveProjective;
+use pairing::Engine;
+
+use super::{Parameters, VerifyingKey};
+use crate::{Circuit, SynthesisError};
+use crate::domain::Domain;
+use crate::multicore::Worker;
+use crate::error::Result;
+
+mod assembly;
+use assembly::*;
+
+/// Generates a random common reference string for
+/// a circuit.
+pub fn generate_random_parameters<E,C,R>(circuit: C, rng: &mut R) -> Result<Parameters<E>>
```

Comment:
Can this be renamed generate_random_crf

Reply:
This is an important API function and so I’ve avoided changing it until a later point, but I’ll note the suggested change.

Reply:
Agreed, following plan to hold off public API changes